packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -135,12 +135,16 @@ export class Arn {
    *      components of the ARN.
    */
   public static parse(arn: string, sepIfToken: string = '/', hasName: boolean = true): ArnComponents {
-    if (Token.isUnresolved(arn)) {
+    const components = arn.split(':') as Array<string | undefined>;
```

Comment:
I would only do that if the string starts with `arn:`

Reply:
Why? It's not likely to throw.